app/controllers/hq/sponsors_controller.rb

```diff
@@ -47,6 +48,12 @@ def load_sponsor
     @sponsor= Sponsor.find(params[:id])
   end
 
+  def load_active_and_inactive_sponsorships
+    sponsorships = Sponsorship.where("sponsor_id = ?", @sponsor.id).eager_load(:orphan, :sponsor).all
```

Comment:
My only concern with the `eager` approach is with time consumption. But it looks nice and simpler than the join approach. If you are interested [this](http://blog.arkency.com/2013/12/rails4-preloading/) is a good article. Nice work cristian :+1: